weave/flow/scorer.py

```diff
@@ -1,12 +1,376 @@
-# Keeping this file for now to avoid breaking changes.
-# In future, users should import all scoring functionality from weave.scorers
-import warnings
-
-from weave.scorers import *
-
-warnings.warn(
-    "Importing from weave.flow.scorer is deprecated. "
-    "Please import from weave.scorers in the future.",
-    DeprecationWarning,
-    stacklevel=2,
-)
+import inspect
+import textwrap
+from collections import defaultdict
+from collections.abc import Sequence
+from dataclasses import dataclass
+from numbers import Number
+from typing import Any, Callable, Optional, Union, cast
+
+import numpy as np
+from pydantic import BaseModel, Field
+
+import weave
+from weave.flow.obj import Object
+from weave.trace.errors import OpCallError
+from weave.trace.isinstance import weave_isinstance
+from weave.trace.op import Op, as_op, is_op
+from weave.trace.op_caller import async_call_op
+from weave.trace.weave_client import Call, sanitize_object_name
+
+
+class Scorer(Object):
```

Comment:
You can move this here, but it might be cleaner to keep in the `scorers` submodule.

If you want users to still be able to do `weave.Scorer`, then this should work:

```
try:
    from weave.scorers import Scorer
except something:
    Scorer = None  # or an object that tells the user to pip install the scorers extra
```

Reply:
Also, if you're going to do this refactor, can you please split it out into a separate PR?